blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java

```diff
@@ -98,12 +97,36 @@ public void createFilesystem(AttachedBlockDevice attachedDevice, FilesystemOptio
             String driveLetterParam = Strings.isNullOrEmpty(driveLetter) ? "-AssignDriveLetter" : "-DriveLetter " + driveLetter;
             String volumeLabel = filesystemOptions.getVolumeLabel();
             String volumeLabelParam = Strings.isNullOrEmpty(volumeLabel) ? "datadisk" : volumeLabel;
-            WinRmToolResponse response = ((WinRmMachineLocation)machine).executePsScript("Get-Disk | Where partitionstyle -eq 'raw' | " +
+            StringBuilder builder = new StringBuilder();
+            String command =
+                    "do {" +
+                    "  $job = Get-Disk -AsJob; " +
+                    "  Wait-Job $job; " +
+                    "  $disks = Receive-Job -Job $job -Wait; " +
                     // AWS May create an instance store volume, which we need to exclude
                     // See https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/ec2-windows-volumes.html#instance-store-volume-map
-                    "Where {$_.SerialNumber -As [int] -NotIn 78..89 -Or $_.FriendlyName -Ne \"AWS PVDISK\"} | " +
-                    "Initialize-Disk -PartitionStyle MBR -PassThru | New-Partition " + driveLetterParam + " -UseMaximumSize | " +
-                    "Format-Volume -FileSystem " + filesystemOptions.getFilesystemType() +" -NewFileSystemLabel \"" + volumeLabelParam + "\" -Confirm:$false");
+                    "  $rawDisk = $disks | Where PartitionStyle -eq 'RAW' | Where {$_.SerialNumber -As [int] -NotIn 78..89 -Or $_.FriendlyName -Ne \"AWS PVDISK\"} | Select-Object -first 1; " +
```

Comment:
It looks like the old version of this code could support multiple additional disks, but the new version will only support one disk. Is this a regression - did it use to be possible for the user to add multiple additional disks? Is this a configuration we are supporting?

Reply:
The old version was in error - it was doing part of the processing on two disks at the same time, whereas this method is called for each additional volume so should only process on at a time. This assumes (hopefully safely) that the volumes returned by the `Get-Disk` scriptlet are in the same order as the volumes in the YAML. Unfortunately there's no way to identify which is which other than the order in which they appear. I've manually run a `Get-Disk | Format-List *`, and there are no properties that can tie them back to the YAML